src/main/java/DukeException.java

```diff
@@ -0,0 +1,13 @@
+public abstract class DukeException extends Exception {
+
+    public DukeException(String message) {
+        super(message);
+    }
+
+    public abstract String getExceptionType();
+
+    @Override
+    public String toString() {
+        return this.getExceptionType() + " detected! " + super.getMessage();
```

Comment:
A neater way would be to use `String::format`, so that you declaratively show the format of the output `String`.

```java
return String.format(
    "%s detected! %s", ...
);
```